tests/app/celery/test_twilio_tasks.py

```diff
@@ -0,0 +1,116 @@
+from datetime import datetime, timedelta, timezone
+
+from unittest.mock import patch
+
+import pytest
+
+from app.celery.exceptions import NonRetryableException
+from app.celery.twilio_tasks import _get_notifications, update_twilio_status
+from app.constants import (
+    NOTIFICATION_CREATED,
+    NOTIFICATION_SENDING,
+    NOTIFICATION_SENT,
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_TECHNICAL_FAILURE,
+    NOTIFICATION_TEMPORARY_FAILURE,
+    NOTIFICATION_PERMANENT_FAILURE,
+    NOTIFICATION_PREFERENCES_DECLINED,
+)
+
+
+@pytest.mark.parametrize(
+    'status, expected',
+    [
+        (NOTIFICATION_CREATED, True),
+        (NOTIFICATION_SENDING, True),
+        (NOTIFICATION_SENT, False),
+        (NOTIFICATION_DELIVERED, False),
+        (NOTIFICATION_TECHNICAL_FAILURE, False),
+        (NOTIFICATION_TEMPORARY_FAILURE, False),
+        (NOTIFICATION_PERMANENT_FAILURE, False),
+        (NOTIFICATION_PREFERENCES_DECLINED, False),
+    ],
+)
+def test__get_notifications_statuses(sample_notification, status, expected):
+    """Test that _get_notifications() returns either a list with the test notification, or an empty list, depending
+    on the parametrized status. If the status is in the NOTIFICATION_STATUS_TYPES_COMPLETED list, the notification is
+    not returned."""
+    created_at = datetime.now(timezone.utc) - timedelta(minutes=90)
+    notification = sample_notification(created_at=created_at, status=status, sent_by='twilio')
+
+    notifications = _get_notifications()
+    notification_ids = [n.id for n in notifications]
+    if expected:
+        assert notification.id in notification_ids
+    else:
+        assert notification.id not in notification_ids
+
+
+@pytest.mark.parametrize(
+    'minute_offset, expected',
+    [
+        (5, False),
+        (45, False),
+        (90, True),
+        (180, True),
+    ],
+)
+def test__get_notifications_datefilter(sample_notification, minute_offset, expected):
+    """Test that _get_notifications() returns either a list with the test notification, or an empty list, depending
+    on the parametrized minute_offset. If the notification was created more than one hour ago, it is not returned."""
```

Comment:
I thought we wanted to only fetch notification created earlier than one hour ago?

Reply:
Oops, that comment is backwards.